lib/mirage_configure.ml

```diff
@@ -88,6 +95,106 @@ let configure_makefile ~no_depext ~opam_name =
       R.ok ())
     "Makefile"
 
+let custom_runtime = function
+  | #Mirage_key.mode_xen -> Some "xen"
+  | #Mirage_key.mode_unix | #Mirage_key.mode_solo5 -> None
+
+let output_of_target = function
+  | #Mirage_key.mode_unix -> "main.exe"
+  | #Mirage_key.mode_xen | #Mirage_key.mode_solo5 -> "main.exe.o"
+
+let variant_of_target = function
+  | #Mirage_key.mode_solo5 -> "freestanding"
+  | #Mirage_key.mode_xen -> "xen"
+  | #Mirage_key.mode_unix -> "unix"
+
+let exclude_modules_of_target = function
+  | #Mirage_key.mode_solo5 -> [ "config"; "manifest" ]
+  | #Mirage_key.mode_xen | #Mirage_key.mode_unix -> [ "config" ]
+
+let dune_path = Fpath.v "dune.build"
+let dune_project_path = Fpath.v "dune-project"
+let dune_workspace_path = Fpath.v "dune-workspace"
+
+let configure_dune i =
+  let ctx = Info.context i in
+  let warn_error = Key.(get ctx warn_error) in
+  let target = Key.(get ctx target) in
+  let custom_runtime = custom_runtime target in
+  let libs = Info.libraries i in
+  let name = Info.name i in
+  let binary_location = output_of_target target in
+  let cflags =
+    [ "-g"; "-w"; "+A-4-41-42-44"; "-bin-annot"; "-strict-sequence";
+      "-principal"; "-safe-string" ]
+    @ (if warn_error then [ "-warn-error"; "+1..49" ] else [])
+    @ (if terminal () then [ "-color"; "always" ] else [])
+    @ (match custom_runtime with Some runtime -> [ "-runtime-variant"; runtime ] | _ -> []) in
+
+  let s_output_mode = match target with
+    | #Mirage_key.mode_unix -> "(native exe)"
+    | #Mirage_key.mode_solo5 | #Mirage_key.mode_xen -> "(native object)" in
+  let s_libraries = String.concat ~sep:" " libs in
+  let s_cflags = String.concat ~sep:" " cflags in
+  let s_variants = variant_of_target target in
+  let s_forbidden = match target with
+    | #Mirage_key.mode_xen | #Mirage_key.mode_solo5 -> "(forbidden_libraries unix)"
+    | #Mirage_key.mode_unix -> "" in
+  let s_exclude_modules = String.concat ~sep:" " (exclude_modules_of_target target) in
+  let res =
+    Fmt.strf
+{sexp|
+(executable
+  (name main)
+  (modes %s)
+  (libraries %s)
+  (link_flags (:include libs.sexp))
+  (modules (:standard \ %s))
+  (flags %s)
+  %s
+  (variants %s))
+|sexp}
+        s_output_mode
+        s_libraries
+        s_exclude_modules
+        s_cflags
+        s_forbidden
+        s_variants in
+  configure_post_build_rules i ~name ~binary_location ~target >>= fun rules ->
+  let res = res ^ "\n" in
+  let rules = List.map (fun x -> Sexplib.Sexp.to_string_hum x ^ "\n") rules in
+  Bos.OS.File.write_lines dune_path (res :: rules)
+
+let configure_dune_workspace i =
+  let ctx = Info.context i in
+  let target = Key.(get ctx target) in
+  let lang = sexp_of_fmt {sexp|(lang dune 2.0)|sexp}
```

Comment:
can you write all of these lines in one big `{sexp|...|sexp}` statements instead of splitting in multiple lines?